docs/source/api/usb.rst

```diff
@@ -0,0 +1,606 @@
+#######
+USB API
+#######
+
+.. note:: This feature is only supported by ESP32-S2.
```

Comment:
I would suggest to reword this a bit, since very soon we will be adding support to S3 and all mentions for S2 will have to be changed. Maybe something like `This feature is only supported on ESP chips that have USB peripheral, like the ESP32-S2 and ESP32-S3. Some chips, like the ESP32-C3 include native CDC+JTAG peripheral that is not covered here`